packages/babel-parser/src/plugins/typescript/index.ts

```diff
@@ -556,10 +556,16 @@ export default (superClass: ClassWithMixin<typeof Parser, IJSXParserMixin>) =>
       this.expect(tt.parenL);
       if (!this.match(tt.string)) {
         this.raise(TSErrors.UnsupportedImportTypeArgument, this.state.startLoc);
+        // Consume as a primary expression so that we can recover from this error
+        node.argument = super.parseExprAtom() as any;
+      } else {
+        if (process.env.BABEL_8_BREAKING) {
+          node.argument = this.tsParseLiteralTypeNode();
```

Comment:
For invalid things like ``import(`abc`)``, would it be better to parse the argument as a `tsParseLiteralTypeNode` rather than an expression?

Reply:
That was my first thought, but then I realized that the `tsParseLiteralTypeNode` does not handle template literals. It seems that typescript-eslint adopted this approach, too. I will revise this PR.